lib/tapioca/runtime/reflection.rb

```diff
@@ -174,11 +174,15 @@ def resolve_loc(locations)
         resolved_loc.absolute_path || ""
       end
 
-      sig { params(singleton_class: Module).returns(T.nilable(Module)) }
+      sig { params(singleton_class: Class).returns(T.nilable(Module)) }
       def attached_class_of(singleton_class)
-        # https://stackoverflow.com/a/36622320/98634
-        result = ObjectSpace.each_object(singleton_class).find do |klass|
-          singleton_class_of(T.cast(klass, Module)) == singleton_class
+        result = if singleton_class.respond_to?(:attached_object)
```

Comment:
Can we move the check to top level, so that we don't have to keep doing the same check over and over again?

For example, if we define:
```ruby
CLASS_ATTACHED_OBJECT = Class.method_defined?(:attached_object)
```
outside this method, then we can just check that in here:
```suggestion
        result = if CLASS_ATTACHED_OBJECT
```

An even better way of doing it would be to change the method definition based on that top-level feature check like the following:
```ruby
CLASS_ATTACHED_OBJECT = Class.method_defined?(:attached_object)

if CLASS_ATTACHED_OBJECT
  def attached_class_of(singleton_class)
    singleton_class.attached_object
  end
else
  def attached_class_of(singleton_class)
    # Old definition here...
  end
end
```
since you only do the feature detection and the check for it once ever in the runtime of the program.

I am afraid that conditional method definition confuses Sorbet, though.